### PR TITLE
Log CMOR check and generic fix output to separate file

### LINKED ISSUE #2351: Do we really need CMOR warnings?
**Is your feature request related to a problem? Please describe.**

For some "problems", our CMOR checker and automatic fixes raise warnings in the log (for example, missing attribute `positive`, changed long name, etc.). If a lot of models are processed, this may lead to A LOT OF warnings. Most of them have zero relevance for the user. Here is an example with ~20'000 lines of warnings: [main_log.txt](https://github.com/ESMValGroup/ESMValCore/files/14418196/main_log.txt) (basically the second half of the file), e.g.

```bash
...
WARNING [266770] Long name changed from 'Net Downward Flux at Top of Model' to 'Net Downward Radiative Flux at Top of Model'
(for file /work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/NOAA-GFDL/GFDL-CM4/historical/r1i1p1f1/Amon/rtmt/gr1/v20180701/rtmt_Amon_GFDL-CM4_historical_r1i1p1f1_gr1_195001-201412.nc)
WARNING [266770] There were warnings in variable rtmt:
 rtmt: attribute positive not present
loaded from file /work/bd0854/DATA/ESMValTool2/CMIP6_DKRZ/CMIP/NOAA-GFDL/GFDL-CM4/historical/r1i1p1f1/Amon/rtmt/gr1/v20180701/rtmt_Amon_GFDL-CM4_historical_r1i1p1f1_gr1_195001-201412.nc
WARNING [266770] There were warnings in variable rtmt:
 rtmt: attribute positive not present
loaded from file 
WARNING [266767] There were warnings in variable rsus:
 rsus: attribute positive not present
...
```

Let's be honest. No one reads these warnings, and nobody's gonna implement fixes for 20 variables of 50 models just because an attribute is missing.

Thus, I propose to convert all of these warnings into debug messages. This will make the output of the tool MUCH nicer. If anyone wants to look up the problems, they can look in the debug log. Serious problems in the data should raise proper exceptions instead of warnings, anyway.

@ESMValGroup/esmvaltool-coreteam thoughts?

### PR DESCRIPTION
**Pull request by schlunma**
## Description

This PR adds filters to our logging configuration so that output from our CMOR checks and generic fixes will not appear in `stdout` and `main_log.txt` anymore, but rather in a dedicated file `cmor_log.txt`. This makes the output of the tool much cleaner.

Closes #2351

***

## [Before you get started](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#getting-started)

- [x] [☝ Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] Code is written according to the [code quality guidelines](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#code-quality)
- [x] [🧪][2] and [🛠][1] [Documentation](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#documentation) is available
- [x] [🛠][1] [Unit tests](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#tests) have been added
- [x] [🛠][1] Changes are [backward compatible](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#backward-compatibility)
- [x] [🛠][1] Any changed [dependencies have been added or removed](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#dependencies) correctly
- [x] [🛠][1] The [list of authors](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#list-of-authors) is up to date
- [x] [🛠][1] ~All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful~ The Filter class needs to be setup like this, so there is no way to fix the remaining Codacy issue

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository

### CODE REVIEW

**Comment by schlunma** [comment]
Questions I have before proceeding with this:

1. Should `main_log_debug.txt` still contain everything, i.e., also the CMOR-related messages? I think it should.
2. Should the `cmor_log.txt` also include CMOR-related debug messages or only warnings? I think it should only include warnings.

@ESMValGroup/technical-lead-development-team

**Comment by valeriupredoi** [comment]
> Should the cmor_log.txt also include CMOR-related debug messages or only warnings? I think it should only include warnings.

I reckon only warnings is enough - we don't want to fix CMOR, we want to fix our own CMOR-related issues :+1: 

Cheers for doing this, bud! Lemme know when you ready for a revue :fr:

**Comment by schlunma** [comment]
> Cheers for doing this, bud! Lemme know when you ready for a revue 🇫🇷

It is now 😁

File: esmvalcore/config/_config_object.py [hunk 1]
```diff
@@ -347,6 +348,11 @@ def main_log_debug(self):
         """Return main log debug file."""
         return self.session_dir / self.relative_main_log_debug
 
+    @property
+    def cmor_log(self):
+        """Return main log file."""
```
Comment (valeriupredoi) [inline esmvalcore/config/_config_object.py:353]:
```suggestion
        """Return CMOR log file."""
```

File: esmvalcore/config/config-logging.yml [hunk 2]
```diff
@@ -10,24 +10,42 @@ formatters:
...
+    filters: [no_cmor]
   debug_log_file:
     class: logging.FileHandler
     level: DEBUG
     formatter: debug
     filename: main_log_debug.txt
     mode: w
+  cmor_log:  # only contains output from CMOR check and generic fixes
+    class: logging.FileHandler
+    level: DEBUG
```
Comment (valeriupredoi) [inline esmvalcore/config/config-logging.yml:44]: maybe INFO or WARNING as default? See me comment in the PR about this

**Review by valeriupredoi — APPROVED** [review]
many thanks, Manu! If I understand correctly, we decided the CMOR log gets outputted always, not only when the main tool is run in logging DEBUG, right?

**Comment by schlunma** [comment]
> many thanks, Manu! If I understand correctly, we decided the CMOR log gets outputted always, not only when the main tool is run in logging DEBUG, right?

That `DEBUG` here means that all messages (debug and above) are logged to that file. It will always be written, regardless of the user's log settings.

This directly relates to my question above:

> 1. Should `main_log_debug.txt` still contain everything, i.e., also the CMOR-related messages? I think it should.

If I set this to `WARNING`, then only warning messages and above are logged. Is this what we want? I guess yes? Note: the debug messages will always be in the main_log_debug.txt file.

File: esmvalcore/config/config-logging.yml (hunk 2 above)
Comment (schlunma) [reply esmvalcore/config/config-logging.yml]: Done in https://github.com/ESMValGroup/ESMValCore/pull/2361/commits/1b5974a1cd50ce9099a2a879d0f151c897548fdf

**Comment by valeriupredoi** [comment]
> That DEBUG here means that all messages (debug and above) are logged to that file. It will always be written, regardless of the user's log settings.

Gotcha! Good call! :beers: 

And yes, I'd argue only CMOR WARNING messages be written to it - but what do others think about it, am looking at @sloosvel @ehogan and @bouweandela specifically :beer:

File: esmvalcore/experimental/recipe_output.py [hunk 3]
```diff
@@ -247,8 +247,9 @@ def _log_ssh_html_info(self):
         )
         logger.info(
             "It looks like you are connected to a remote machine via SSH. To "
-            "show the output html file, you can try the following command:"
-            "\n%s\nThen visit http://localhost:%s in your browser",
+            "show the output html file, you can try the following command on "
+            "your local machine:\n%s\nThen visit http://localhost:%s in your "
+            "browser",
```
Comment (valeriupredoi) [inline esmvalcore/experimental/recipe_output.py:252]: :smiley:

**Comment by ehogan** [comment]
> > That DEBUG here means that all messages (debug and above) are logged to that file. It will always be written, regardless of the user's log settings.
> 
> Gotcha! Good call! 🍻
> 
> And yes, I'd argue only CMOR WARNING messages be written to it - but what do others think about it, am looking at @sloosvel @ehogan and @bouweandela specifically 🍺

Just to check my understanding: it will be possible to (always) see all CMOR logs via `main_log_debug.txt`, but only warnings will be shown via `cmor_log.txt` and `main_log.txt`? If so, that seems reasonable to me! 👍

**Comment by schlunma** [comment]
> Just to check my understanding: it will be possible to (always) see all CMOR logs via `main_log_debug.txt`, but only warnings will be shown via `cmor_log.txt` and `main_log.txt`? If so, that seems reasonable to me! 👍

CMOR warnings would not be shown in `main_log.txt` (only in `cmor_log.txt`), apart from that that's correct!

**Comment by valeriupredoi** [comment]
@bouweandela would you mind doing a final sweep and merge if all's good in yer books, plss, bud :beer:

**Pull request merged**